compiler/rustc_middle/src/mir/mono.rs

```diff
@@ -490,15 +490,18 @@ impl CodegenUnitNameBuilder<'tcx> {
             // local crate's ID. Otherwise there can be collisions between CGUs
             // instantiating stuff for upstream crates.
             let local_crate_id = if cnum != LOCAL_CRATE {
-                let local_crate_disambiguator = format!("{}", tcx.crate_disambiguator(LOCAL_CRATE));
-                format!("-in-{}.{}", tcx.crate_name(LOCAL_CRATE), &local_crate_disambiguator[0..8])
+                let local_stable_crate_id = tcx.sess.local_stable_crate_id();
+                format!(
+                    "-in-{}.{:08x}",
```

Comment:
It looks like this doesn't actually truncate the hash to 8 characters, thus making it effectively longer. Since these names end up in filenames where length can matter, let's try to keep them at the same length as before (e.g. by masking off the higher bits).